### PR TITLE
satellite/accounting: separate usage and bandwidth limit

### PR DESCRIPTION
**Pull request by stefanbenten**
Should go in after #3877 

What: separate the usage and bandwidth limit to allow more granular project limit increase handling.

Why: Treating them as one is very weird and clunky.

Please describe the tests:
 - Test 1: adjusted
 - Test 2: NA
 
Please describe the performance impact: slightly more DB reads.

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/satellitedb/migrate.go [hunk 1]
```diff
@@ -1067,6 +1067,36 @@ func (db *satelliteDB) PostgresMigration() *migrate.Migration {
...
+			{
+				DB:          db.DB,
+				Description: "add separate bandwidth column",
+				Version:     107,
+				Action: migrate.Func(func(ctx context.Context, log *zap.Logger, db tagsql.DB, tx tagsql.Tx) error {
+					// ensure migration is not flattened and executed separate from 108
+					_, err := tx.Exec(ctx,
+						`ALTER TABLE projects ADD COLUMN bandwidth_limit bigint NOT NULL DEFAULT 0;`,
+					)
+					if err != nil {
```
Comment (mniewrzal) [inline satellite/satellitedb/migrate.go:1079]: can be just `return ErrMigrate.Wrap(err)`

File: satellite/satellitedb/migrate.go [hunk 2]
```diff
@@ -1067,6 +1067,36 @@ func (db *satelliteDB) PostgresMigration() *migrate.Migration {
...
+			},
+			{
+				DB:          db.DB,
+				Description: "backfill bandwidth column with previous limits",
+				Version:     108,
+				Action: migrate.Func(func(ctx context.Context, log *zap.Logger, db tagsql.DB, tx tagsql.Tx) error {
+					_, err := tx.Exec(ctx,
+						`UPDATE projects SET bandwidth_limit = usage_limit;`,
+					)
+					if err != nil {
```
Comment (mniewrzal) [inline satellite/satellitedb/migrate.go:1093]: can be just `return ErrMigrate.Wrap(err)`

File: satellite/satellitedb/migrate.go [hunk 3]
```diff
@@ -1067,6 +1067,36 @@ func (db *satelliteDB) PostgresMigration() *migrate.Migration {
 					ON CONFLICT(project_id, interval_month) DO UPDATE SET egress_allocated = EXCLUDED.egress_allocated::bigint;`,
 				},
 			},
+			{
+				DB:          db.DB,
+				Description: "add separate bandwidth column",
+				Version:     107,
+				Action: migrate.Func(func(ctx context.Context, log *zap.Logger, db tagsql.DB, tx tagsql.Tx) error {
+					// ensure migration is not flattened and executed separate from 108
```
Comment (mniewrzal) [inline satellite/satellitedb/migrate.go:1075]: can you say something more why it cannot be as one migration step?

File: satellite/satellitedb/dbx/satellitedb.dbx [hunk 4]
```diff
@@ -289,6 +289,7 @@ model project (
     field name           text
     field description    text      ( updatable )
     field usage_limit    int64     ( updatable, default 0 )
+    field bandwidth_limit   int64     ( updatable, default 0 )
```
Comment (mniewrzal) [inline satellite/satellitedb/dbx/satellitedb.dbx:292]: nit: can you adjust indentations for old fields?

File: satellite/accounting/db.go [hunk 5]
```diff
@@ -175,6 +175,8 @@ type ProjectAccounting interface {
 	GetStorageTotals(ctx context.Context, projectID uuid.UUID) (int64, int64, error)
 	// UpdateProjectUsageLimit updates project usage limit.
 	UpdateProjectUsageLimit(ctx context.Context, projectID uuid.UUID, limit memory.Size) error
+	// UpdateProjectUsageLimit updates project usage limit.
```
Comment (mniewrzal) [inline satellite/accounting/db.go:178]: a comment is copy pasted

File: satellite/satellitedb/migrate.go (hunk 3 above)
Comment (stefanbenten) [reply satellite/satellitedb/migrate.go:1075]: Cockroach cannot add a column and set the entries in one go.  It have to be seperate steps.
In addition it has to be done in a migrate.Func as our flattenMigration function instead would again condense them into one.

**Pull request merged**